test/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDaoTest.java

```diff
@@ -206,34 +295,81 @@ public void versioningASurveyCopiesTheQuestions() {
     @Test
     public void canPublishASurvey() {
         Survey survey = surveyDao.createSurvey(testSurvey);
-        survey = surveyDao.publishSurvey(survey);
+        surveysToDelete.add(new GuidCreatedOnVersionHolderImpl(survey));
+        survey = surveyDao.publishSurvey(studyIdentifier, survey);
+        schemaIdsToDelete.add(survey.getIdentifier());
 
         assertTrue("Survey is marked published", survey.isPublished());
 
-        Survey pubSurvey = surveyDao.getSurveyMostRecentlyPublishedVersion(studyIdentifier, survey.getGuid());
+        // validate the corresponding schema was created
+        UploadSchema uploadSchema = uploadSchemaDao.getUploadSchema(TEST_STUDY_IDENTIFIER, survey.getIdentifier());
+        int schemaRev = uploadSchema.getRevision();
+
+        assertEquals(survey.getIdentifier(), uploadSchema.getSchemaId());
+        assertEquals(survey.getSchemaRevision().intValue(), uploadSchema.getRevision());
+        assertEquals(survey.getName(), uploadSchema.getName());
+        assertEquals(UploadSchemaType.IOS_SURVEY, uploadSchema.getSchemaType());
+
+        List<UploadFieldDefinition> fieldDefList = uploadSchema.getFieldDefinitions();
+        assertEquals(9, fieldDefList.size());
+
+        assertEquals("high_bp", fieldDefList.get(0).getName());
+        assertEquals(UploadFieldType.BOOLEAN, fieldDefList.get(0).getType());
 
+        assertEquals("last_checkup", fieldDefList.get(1).getName());
+        assertEquals(UploadFieldType.CALENDAR_DATE, fieldDefList.get(1).getType());
+
+        assertEquals("last_reading", fieldDefList.get(2).getName());
+        assertEquals(UploadFieldType.TIMESTAMP, fieldDefList.get(2).getType());
+
+        assertEquals("deleuterium_dosage", fieldDefList.get(3).getName());
+        assertEquals(UploadFieldType.FLOAT, fieldDefList.get(3).getType());
+
+        assertEquals("bp_x_day", fieldDefList.get(4).getName());
+        assertEquals(UploadFieldType.INT, fieldDefList.get(4).getType());
+
+        assertEquals("time_for_appt", fieldDefList.get(5).getName());
+        assertEquals(UploadFieldType.STRING, fieldDefList.get(5).getType());
```

Comment:
Parenthetically, I would call the UploadFieldDefinition name an identifier. Sometimes, these things also have "labels" which can be changed for UI presentations without changing the identifier. So name ends up being a little ambiguous.